### ISSUE TITLE
compatibility with luks

### ISSUE THREAD

**Issue by auipga** [error10/kexec-reboot#5]
Hi there, I really like your solution. Can you please provide the additional steps to get this working with luks encryption enabled? Thank you.

**Comment by error10** [comment]
What do you mean by "additional steps"? There are no additional steps to use this with LUKS. Are you having some sort of issue?

**Comment by auipga** [comment]
Then the reason for the script not working is a different one.
On a fresh installation of ArchLinux is gives me:
```sh
Kernel --class is not readable
Couldn't find a grub configuration anywhere!
```
Looks like the regex part doesn't work correctly for my `grub.cfg`.  Here is my menuentry:
```sh
### BEGIN /etc/grub.d/10_linux ###
menuentry 'ArchEnc Linux' --class archenc --class gnu-linux --class gnu --class os $menuentry_id_option 'gnulinux-simple-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' {
	load_video
	set gfxpayload=keep
	insmod gzio
	insmod part_msdos
	insmod fat
	set root='hd4,msdos1'
	if [ x$feature_platform_search_hint = xy ]; then
	  search --no-floppy --fs-uuid --set=root --hint-bios=hd4,msdos1 --hint-efi=hd4,msdos1 --hint-baremetal=ahci4,msdos1  XXXX-XXXX
	else
	  search --no-floppy --fs-uuid --set=root XXXX-XXXX
	fi
	echo	'Loading Linux linux ...'
	linux	/vmlinuz-linux root=UUID=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx rw rootflags=subvol=ROOT cryptdevice=/dev/disk/by-uuid/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:root root=/dev/mapper/root quiet
	echo	'Loading initial ramdisk ...'
	initrd  /initramfs-linux.img
}

# skipped some more entries

### END /etc/grub.d/10_linux ###
```
I managed to get at least the correct `/vmlinuz-linux` but I've overridden that file :(

**Comment by auipga** [comment]
Steps to make the detection work for me:
* Allow short UUIDs for FAT partitions
`[0-9A-F]{4}-[0-9A-F]{4}`
* Change space `\s` to tab `\t` after "linux" because the name of my preset is also "linux" followed by a space and 3 dots
`linux(efi|16)?\t`

**Comment by auipga** [comment]
Just rebooted my HP ProLiant within seconds. Wow, this is amazing!!

**Comment by error10** [comment]
Fixed in v0.8.3. Thank you for the report!
